js/ui/searchbar.js

```diff
@@ -0,0 +1,178 @@
+
+        chrome.fileSystem.getDisplayPath(node.entry, function(path) {
+          var file = new File(node.entry);
+            if (matches >= maxMatches) {
+              return;
+            }
+
+            file.read(function(err, data) {
+              var lines = data.split("\n");
+              var line, msg;
+              var firstFind = true;
+              var printedLines = {}; // only print each line once per file per search
+
+              for (var i = 0; i < lines.length && matches < maxMatches; i++) {
+                compareLine = isCaseSensitive ? lines[i] : lines[i].toUpperCase();
+                if (compareLine.indexOf(searchQuery) > -1) {
+                  matches++;
+                  msg = "";
+                  if (!printedLines[i] && !printedLines[i-1]) { // only add break if it and the line before it hace not been printed
+                    if (firstFind) {
+                      msg += "\n" + node.entry.fullPath + "\n";
+                      firstFind = false;
+                    } else {
+                      msg += "...\n";
+                    }
+                    msg += self.formatResultCode(i-1, prevLine);
+                    msg += self.formatResultCode(i, lines[i]);
+                    printedLines[i] = true;
+                  }
+
+                  if (i < lines.length - 1) {
+                    msg += self.formatResultCode(i+1, lines[i+1]);
+                    printedLines[i+1] = true;
+                  }
+                  self.appendToResults(resultsTab, msg);
+                }
+                prevLine = lines[i];
+              }
+
+              if (--filesToSearch === 0) { // we have finished searching all files
+                self.appendToResults(resultsTab, "\n\n" + matches + " matches found.");
+                if (matches >= maxMatches) {
+                  self.appendToResults(resultsTab, "\nSearch was cancelled. To allow search to return more matches, see User Preferences.")
+                }
+              }
+            });
+        });
+      }
+
+      // search through project directories
+      project.directories.forEach(function(directoryNode) {
+        searchDirectory(directoryNode);
+      });
+    },
+
+    formatResultCode: function(lineNumber, code) {
+      return "  " + (lineNumber+1) + ": " + code + "\n";
+    },
+
+    appendToResults: function(resultsTab, text) {
+      if (text === "") {
+        return;
+      }
+      var insertRow = resultsTab.doc.getLength();
+      resultsTab.doc.insert({row: insertRow, column: 0}, text);
+    },
+
+    activate: function(mode) {
+      var highlighted = editor.getSelectedText();
+      if (highlighted) {
+        this.input.value = highlighted;
+      }
+
+      this.element.addClass("active");
+      this.input.focus();
+    },
+    
+    deactivate: function() {
+      this.element.removeClass("active");
+    }
+  };
+
+  var searchbar = new Searchbar();
+
+  command.on("searchbar:show-project-search", function() {
```

Comment:
We should also register for the `init:restart` event, which fires whenever the settings change. That way, we can update the maxMatches value without having to restart the whole app.